timesketch/lib/analyzers/feature_plugins/winevt.py

```diff
@@ -0,0 +1,172 @@
+"""This file contains the plugin for Windows event logs feature extraction."""
+
+from typing import List
+
+import logging
+
+from timesketch.lib.analyzers import interface as base_interface
+from timesketch.lib.analyzers.feature_plugins import interface
+from timesketch.lib.analyzers.feature_plugins import manager
+
+logger = logging.getLogger("timesketch.analyzers.feature")
+
+
+class WindowsEventFeatureExtractionPlugin(interface.BaseFeatureExtractionPlugin):
+    """A plugin for Windows event log feature extraction."""
+
+    NAME = "winevt_feature_extraction"
+    DISPLAY_NAME = "Windows Event Log Feature Extraction"
+    DESCRIPTION = (
+        "This plugin extracts Windows event logs attributes from plaso output"
+        " attribute `strings`"
+    )
+
+    EVENT_FIELDS = ["strings"]
+
+    def run_plugin(self) -> str:
+        """Initializes plugin class object.
+
+        Returns:
+            str: Summary of plugin output.
+        """
+
+        # Stores extraction summary for each feature.
+        results = []
+
+        feature_kwargs = self.load_feature_config()
+        if not feature_kwargs:
+            return f"No feature configuration data for {self.NAME}"
+
+        for feature in feature_kwargs:
+            # NOTE: feature_name and feature_config value checking is done at the sink
+            # function `extract_features`.
+            feature_name = feature.get("feature", None)
+            feature_config = feature.get("feature_config", None)
+
+            result = self.extract_features(feature_name, feature_config)
+            if not result:
+                logger.debug("No result for plugin run for %s", feature_name)
+            results.append(result)
+
+        return "\n".join(results)
+
+    def extract_features(self, name: str, config: dict) -> str:
+        """Extracts features from events.
+
+        Args:
+            name (str): Features extraction name.
+            config (dict): A dict that contains the configuration fo the
+                feature extraction.
+
+        Returns:
+            str: Returns summary of the feature extraction for `feature_name`.
+        """
+
+        if not name:
+            raise ValueError("Feature name is empty")
+
+        if not config:
+            raise ValueError("Feature configuration value is empty")
+
+        source_name = config.get("source_name", None)
+        if not source_name:
+            raise ValueError("source_name is empyt")
```

Comment:
Would it make sense to check for all mandatory fields at this point and fail if any of them are not defined?

Reply:
Done. Handled in `validate_feature_config`.